components/form/LifecycleHooks.vue

```diff
@@ -0,0 +1,135 @@
+<script>
+import Vue from 'vue';
+
+import HookOption from '@/components/form/HookOption';
+import { _VIEW } from '@/config/query-params';
+
+export default {
+  components: { HookOption },
+
+  props: {
+    mode: {
+      type:     String,
+      required: true,
+    },
+
+    // container spec
+    value: {
+      type:    Object,
+      default: () => {
+        return {};
+      }
+    },
+  },
+
+  data() {
+    return {
+      showHooks:    [],
```

Comment:
The HookOptions don't render correctly when you try to edit a workload with lifecycle hooks defined because this always starts empty. I think you need to check if `value.postStart` and `value.preStop` are defined and add them here to get edit and detail>view config working.

Reply:
HookOption is now checking the container spec for any defined lifecycle hooks when created.
![createdHooks](https://user-images.githubusercontent.com/40806497/134416425-aa3b1843-4760-4319-9bea-819bdcef8f7a.png)